scripts/generate_notification_images.sh

```diff
@@ -0,0 +1,142 @@
+fi
+source "${ALLSKY_HOME}/variables.sh"
+ME="$(basename "${BASH_ARGV0}")"
+
+readonly USAGE="Usage: ${ME} [StartingUp TextColor Font FontSize StrokeColor StrokeWidth BgColor BorderColor Extensions ImageSize 'Message']"
+readonly ALL_EXTS="jpg png"		# all the image filename extensions we support
+
+function make_image() {
+	BASENAME="$1"
+	TEXTCOLOR="${2:-"white"}"
+	FONT="${3:-"Helvetica-Bold"}"
+	FONT_SIZE="${4:-128}"
+	INTERLINE_SPACING=$(echo "${FONT_SIZE} / 3" | bc)
+	STROKE_COLOR="${5:-"black"}"
+	STROKE_WIDTH="${6:-2}"
+	BGCOLOR="${7:-"#404040"}"
+	BORDER_WIDTH="${8:-0}"
+	BORDER_COLOR="${9:-"white"}"
+	EXTS="${10:-"${ALL_EXTS}"}"
+	# IM_SIZE is a hack to make these images more detectable. Typically camera images are
+	# at least an even number of pixels, and usually a multiple of 8 pixels.
+	# So just in case someone has a camera configured for 960x720 images, or
+	# is rescaling to 960x720... this will allow these notification images to be detected.
+	IM_SIZE="${11:-"959x719"}"
+	MSG="${12}"
+
+	echo "${BASENAME}" | grep -qEi "[.](${ALL_EXTS/ /|})"
+	if [ $? -ne 1 ] ; then
+		echo -e "${RED}ERROR: Do not add an extension to the basename${NC}."
+		echo "${USAGE}"
+		exit 1
+	fi
+
+	if [ ${BORDER_WIDTH} -ne 0 ]; then
+		BORDER="-border ${BORDER_WIDTH} -bordercolor ${BORDER_COLOR}"
+	else
+		BORDER=""
+	fi
+
+	echo "Creating '${BASENAME}'"
+	for EXT in ${EXTS} ; do
+		convert \
+			-fill "${TEXTCOLOR}" \
+			-font "${FONT}" \
+			-pointsize "${FONT_SIZE}" \
+			-interline-spacing "${INTERLINE_SPACING}" \
+			-stroke "${STROKE_COLOR}" \
+			-strokewidth "${STROKE_WIDTH}" \
+			-background "${BGCOLOR}" \
+			${BORDER} \
+			-gravity center \
+			-depth 8 \
+			-size "${IM_SIZE}" \
+			label:"${MSG}" \
+			"${BASENAME}.${EXT}"
+	done
+}
+
+which mogrify > /dev/null
+if [ $? -ne 0 ] ; then
+	# Testing for mogrify which seems like a much more distinctive executable
+	# name than "convert". I assume that if "mogrify" is in the path, then
+	# ImageMagick is installed and "convert" will run ImageMagick and not some
+	# other tool.
+	echo -e "${RED}ERROR: ImageMagick does not appear to be installed. Please install it.${NC}" >&2
+	exit 2
+fi
+
+# If the arguments were specified on the command line, use them instead of the list below.
+if [ $# -eq 12 ]; then
+	if [ "${1}" = "" -o "${12}" = "" ]; then
+		echo -e '${RED}ERROR: Basename ($1) and message ($12) must be specified.${NC}' >&2
+		echo "${USAGE}"
+		exit 1
+	fi
+	make_image "${@}"
+
+elif [ $# -eq 0 ]; then
+	#ake_image  1: Basename            2: "TxtColor"                          \
+	#           3: "Font"              4: "FontSize"                          \
+	#		    5: "StrokeColor"       6: "StrokeWidth"                       \
+	#		    7: "BgColor"           8: "BorderWidth"    9: "BorderColor"   \
+	#		   10: "Extensions"       11: "ImageSize"                         \
+	#		   12: "Message"
+	make_image NotRunning                 ""                                  \
```

Comment:
@EricClaeys 
commit 77455880b1a2795044fee69ee4bfe1370f1037ef was much cleaner (for me)
The test was a pain for me: not in one line, ich have to count, remember,......

Reply:
@AndreasLMeg   LOL.  I found it a pain when it was all on one line since things didn't line up and if they did, it went onto multiple lines on my screen which made it hard to read.
The good thing is that we'll rarely need to change the table going forward.
Would it help if we duplicated the header on lines 90-95 between every call to "make_image" (or every other call)?